pkg/kv/postgres/store.go

```diff
@@ -0,0 +1,239 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type Driver struct{}
+
+type Store struct {
+	Pool   *pgxpool.Pool
+	Params *Params
+	closed bool
+}
+
+type Entries struct {
+	rows  pgx.Rows
+	entry *kv.Entry
+	err   error
+}
+
+const (
+	DriverName = "postgres"
+
+	defaultTableName = "kv"
+	paramTableName   = "lakefskv_table"
+)
+
+//nolint:gochecknoinits
+func init() {
+	kv.Register(DriverName, &Driver{})
+}
+
+func (d *Driver) Open(ctx context.Context, name string) (kv.Store, error) {
+	// TODO(barak): name format - we need to accept a connection string to the db + params related to db + params related to kv
+	// TODO(barak): handle reuse - open the same kv - reuse the pool
+	config, err := pgxpool.ParseConfig(name)
```

Comment:
Can it get a config struct like we do elsewhere and work it way from there?

Reply:
It should be dynamic to support all the future possible database implementations